addons/binding/org.openhab.binding.helios/src/main/java/org/openhab/binding/helios/handler/HeliosHandler213.java

```diff
@@ -0,0 +1,942 @@
+                properties.put(BUILD_TYPE, systemInfo.buildType);
+                properties.put(DEVICE_NAME, systemInfo.deviceName);
+                updateProperties(properties);
+            }
+
+            List<RESTSwitch> switches = getSwitches();
+
+            if (switches != null) {
+                for (RESTSwitch aSwitch : switches) {
+                    if (aSwitch.enabled.equals("true")) {
+                        logger.debug("Adding a channel to the Helios IP Vario '{}' for the switch with id '{}'",
+                                getThing().getUID().toString(), aSwitch.id);
+                        ThingBuilder thingBuilder = editThing();
+                        ChannelTypeUID enablerUID = new ChannelTypeUID(BINDING_ID, SWITCH_ENABLER);
+                        ChannelTypeUID triggerUID = new ChannelTypeUID(BINDING_ID, SWITCH_TRIGGER);
+
+                        Channel channel = ChannelBuilder
+                                .create(new ChannelUID(getThing().getUID(), "switch" + aSwitch.id + "active"), "Switch")
+                                .withType(enablerUID).build();
+                        thingBuilder.withChannel(channel);
+                        channel = ChannelBuilder
+                                .create(new ChannelUID(getThing().getUID(), "switch" + aSwitch.id), "Switch")
+                                .withType(triggerUID).build();
+                        thingBuilder.withChannel(channel);
+                        updateThing(thingBuilder.build());
+                    }
+                }
+            }
+
+            List<RESTPort> ports = getPorts();
+
+            if (ports != null) {
+                for (RESTPort aPort : ports) {
+                    logger.debug("Adding a channel to the Helios IP Vario '{}' for the IO port with id '{}'",
+                            getThing().getUID().toString(), aPort.port);
+                    ThingBuilder thingBuilder = editThing();
+                    ChannelTypeUID triggerUID = new ChannelTypeUID(BINDING_ID, IO_TRIGGER);
+
+                    Map<String, String> channelProperties = new HashMap<String, String>();
+                    channelProperties.put("type", aPort.type);
+
+                    Channel channel = ChannelBuilder
+                            .create(new ChannelUID(getThing().getUID(), "io" + aPort.port), "Switch")
+                            .withType(triggerUID).withProperties(channelProperties).build();
+                    thingBuilder.withChannel(channel);
+                    updateThing(thingBuilder.build());
+                }
+            }
+        }
+    };
+
+    protected Runnable resetRunnable = new Runnable() {
+        @Override
+        public void run() {
+            logger.debug("Resetting the Helios IP Vario handler for '{}'", getThing().getUID());
+            dispose();
+            initialize();
+        }
+    };
+
+    protected Runnable postInitRunnable = new Runnable() {
```

Comment:
This is very nasty. If you device is offline, you will block a complete thread from the thread pool.
You should now in your code yourself, when you set the Thing to ONLINE and you could schedule the postInitialize at that moment only.